File: js/58_length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function (s) {

    counter = 0
    flag = false

    i = s.length - 1
    while (i >= 0) {

        // aZ
        if (s[i] !== " ") {
            counter++
            flag = true
        }
        // space
        else {
            if (flag)
                break
        }
        i--
    }

    return counter
};

File: python/58_length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:

        counter = 0
        i = len(s) - 1

        while i >= 0 and s[i] == " ":
            i -= 1

        while i >= 0 and s[i] != " ":
            counter += 1
            i -= 1

        return counter


# Time:  O(n)
# Space: O(1)
